### PR TITLE
Merge MOBILE & WEB servers 

### LINKED ISSUE #211: [ALL] Merge Mobile and Web servers
- Put the Mobile server in the Web server
- Merge the database
- Use the following package when returning HTTP status code: https://www.npmjs.com/package/http-status-codes

### PR DESCRIPTION
**Pull request by H-chauvet**
Merge server mobile into Web server
Test also merge
Do not hesitate to contact @31Nathan if questions
Closes #211

### CODE REVIEW

File: Configurateur/back/middleware/Mobile/jwt.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ */
+function generateToken(id) {
+  return jwt.sign({ id }, process.env.JWT_ACCESS_SECRET, {
+    expiresIn: parseInt(process.env.JWT_EXPIRE)
+  });
+}
+
+module.exports = {
+  generateToken,
+};
```
Comment (tbellicha) [inline Configurateur/back/middleware/Mobile/jwt.js:19]: New line

File: Configurateur/back/middleware/Mobile/jwt.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const jwt = require("jsonwebtoken");
+const crypto = require("crypto");
+
+/**
+ *
```
Comment (tbellicha) [inline Configurateur/back/middleware/Mobile/jwt.js:5]: Remove this line

File: Configurateur/back/routes/Mobile/containers.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    const container = await containerCtrl.getAllContainers();
+
+    return res.status(200).json(container);
+  } catch (err) {
+    next(err);
+  }
+});
+
+router.get('/:containerId', async (req, res, next) => {
+  // try {
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/containers.js:18]: Remove this commented line

File: Configurateur/back/routes/Mobile/contact.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    await contactCtrl.createContact(name, email, message)
+
+    return res.status(201).json({ message: 'contact saved' })
+  } catch (err) {
+    console.error(err.message)
+    return res.status(401).send('Error while saving contact.')
+  }
+})
+
+module.exports = router
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/contact.js:23]: New line

File: Configurateur/back/routes/Mobile/items.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      return res.status(401).json("article not found")
+    }
+    return res.status(200).json(article)
+  } catch (err) {
+    console.error(err.message)
+    return res.status(401).send('An error occurred')
+  }
+})
+
+module.exports = router
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/items.js:29]: New line

File: Configurateur/back/routes/Mobile/containers.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      return res.status(204).json({ message: 'Container doesn\'t have items' })
+    }
+    return res.status(200).json(container.items)
+  } catch (err) {
+    console.error(err.message)
+    return res.status(401).send('An error occurred')
+  }
+})
+
+module.exports = router
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/containers.js:59]: New Line

File: Configurateur/back/routes/Mobile/opinion.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+      return res.status(201).json({ message: 'opinion saved' })
+    } catch (err) {
+      console.error(err.message)
+      return res.status(401).send('An error occurred')
+    }
+  }
+)
+
+module.exports = router
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/opinion.js:72]: New line

File: Configurateur/back/routes/Mobile/rent.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    }
+  }
+)
+
+router.get('/:rentId',
+  passport.authenticate('jwt', { session: false }), async (req, res) => {
+    try {
+      if (!req.user) {
+        return res.status(401).send('Invalid token')
+      }
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/rent.js:72]: You don't check if the user is in the DB ?

File: Configurateur/back/routes/Mobile/rent.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+      return res.status(201).json({ message: 'location returned' })
+    } catch (err) {
+      console.error(err.message)
+      return res.status(401).send('An error occurred')
+    }
+  }
+)
+
+
+module.exports = router
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/rent.js:117]: New line

File: Configurateur/back/routes/Mobile/user.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+      const user = await userCtrl.findUserById(req.params.userId)
+
+      return res.status(200).json({ user });
+    } catch (err) {
+      console.error(err.message)
+      return res.status(401).send('An error occurred.')
+    }
+  }
+)
+/// api/user AND api/user/notifications
```
Comment (tbellicha) [inline Configurateur/back/routes/Mobile/user.js:90]: Missing new line

File: Configurateur/back/tests/mobile/contact.test.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({})
+            .expect(401, callback)
+        },
+      ],
+      done
+    )
+  })
+})
```
Comment (tbellicha) [inline Configurateur/back/tests/mobile/contact.test.js:35]: New line

File: Configurateur/back/tests/mobile/mailVerification.test.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+              .set('Content-Type', 'application/json')
+              .set('Accept', 'application/json')
+              .send({ token: '' })
+              .expect(401, callback)
+          },
+        ],
+        done
+      )
+    })
+});
```
Comment (tbellicha) [inline Configurateur/back/tests/mobile/mailVerification.test.js:20]: New line

**Review by tbellicha — CHANGES_REQUESTED** [review]

File: Application/mobile/app_code/test/article_list_data_test.dart [hunk 13]
```diff
@@ -20,13 +20,9 @@ void main() {
...
       test('ArticleListData constructor should create an instance', () {
         final articleListData = ArticleData(
-            id: 'id',
-            containerId: 'containerId',
-            name: 'name',
-            available: true,
-            price: 3);
-        expect(articleListData.id, 'id');
-        expect(articleListData.containerId, 'containerId');
+            id: 0, containerId: 0, name: 'name', available: true, price: 3);
```
Comment (invpuppet) [inline Application/mobile/app_code/test/article_list_data_test.dart:23]: maybe add a ',' before the end of the ')' for the code standards

File: Application/mobile/app_code/test/article_list_data_test.dart [hunk 14]
```diff
@@ -68,11 +60,7 @@ void main() {
         'ArticleDataCard should show from ArticleData',
         (WidgetTester tester) async {
           final articleListData = ArticleData(
-              id: 'id',
-              containerId: 'containerId',
-              name: 'name',
-              available: true,
-              price: 3);
+              id: 0, containerId: 1, name: 'name', available: true, price: 3);
```
Comment (invpuppet) [inline Application/mobile/app_code/test/article_list_data_test.dart:63]: same as before

File: Application/mobile/app_code/test/confirm_rent_test.dart [hunk 15]
```diff
@@ -18,7 +18,7 @@ void main() {
 
   testWidgets('Confirm rent test', (WidgetTester tester) async {
     final ArticleData data = ArticleData(
-        id:'' , containerId: '', name: 'Ballon', available: true, price: 8);
+        id: 0, containerId: 1, name: 'Ballon', available: true, price: 8);
```
Comment (invpuppet) [inline Application/mobile/app_code/test/confirm_rent_test.dart:21]: same here

File: Configurateur/back/controllers/Common/container.js [hunk 16]
```diff
@@ -49,17 +54,25 @@ exports.updateContainer = (id, container) => {
...
 };
+
+exports.getItemsFromContainer = (containerId) => {
+  return db.Containers.findUnique({
+    where: { id: containerId },
+    select: {
+      items : true
+    },
+  })
+}
```
Comment (invpuppet) [inline Configurateur/back/controllers/Common/container.js:78]: new line

**Review by invpuppet — CHANGES_REQUESTED** [review]
overall it's good for me
just some new lines and things like that to fix

**Review by hugolopespinto — APPROVED** [review]
The mobile part looks good for me.

File: Configurateur/back/routes/Mobile/rent.js (hunk 8 above)
Comment (31Nathan) [reply Configurateur/back/routes/Mobile/rent.js:72]: Done

File: Configurateur/docker-compose.yaml [hunk 17]
```diff
@@ -1,44 +1,8 @@
...
-    restart: always
   web_app:
     build:
       context: ./front
     ports:
-      - 80:5000
-
-volumes:
-  mysql-data:
+      - 80:5000
```
Comment (invpuppet) [inline Configurateur/docker-compose.yaml:8]: new line

File: docker-compose.yaml [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+      database:
+        condition: service_healthy
+    ports:
+      - 3000:3000
+    stdin_open: true
+    tty: true
+    restart: always
+
+volumes:
+  mysql-data:
```
Comment (invpuppet) [inline docker-compose.yaml:39]: new line

File: Server/controllers/Web/contact.js [hunk 19]
```diff
@@ -0,0 +1,8 @@
+const { db } = require('../../middleware/database')
+
+exports.registerMessage = data => {
+    return db.Contact_Web.create({
+      data: data
+    })
+  }
+
```
Comment (invpuppet) [inline Server/controllers/Web/contact.js:8]: new line

**Review by invpuppet — DISMISSED** [review]
some new lines

File: Server/controllers/Web/contact.js (hunk 19 above)
Comment (H-chauvet) [reply Server/controllers/Web/contact.js:8]: Line 8 is already a new line

**Review by CedricCORGE — APPROVED** [review]

**Pull request merged**